hummingbot/client/config/strategy_config_data_types.py

```diff
@@ -0,0 +1,171 @@
+from typing import Dict
+
+from pydantic import Field, validator
+
+from hummingbot.client import settings
+from hummingbot.client.config.config_data_types import BaseClientModel, ClientConfigEnum, ClientFieldData
+from hummingbot.client.config.config_validators import (
+    validate_exchange,
+    validate_market_trading_pair,
+    validate_strategy,
+)
+from hummingbot.client.settings import AllConnectorSettings, ConnectorType
+
+
+class BaseStrategyConfigMap(BaseClientModel):
+    strategy: str = Field(
+        default=...,
+        client_data=ClientFieldData(
+            prompt=lambda mi: "What is your market making strategy?",
+            prompt_on_new=True,
+        ),
+    )
+
+    @validator("strategy", pre=True)
+    def validate_strategy(cls, v: str):
+        ret = validate_strategy(v)
+        if ret is not None:
+            raise ValueError(ret)
+        return v
+
+
+class BaseTradingStrategyConfigMap(BaseStrategyConfigMap):
+    exchange: ClientConfigEnum(  # rebuild the exchanges enum
+        value="Exchanges",  # noqa: F821
+        names={e: e for e in
+               sorted([connector_setting.name for connector_setting
+                       in AllConnectorSettings.get_connector_settings().values()
+                       if connector_setting.type is ConnectorType.Exchange] + settings.PAPER_TRADE_EXCHANGES)},
+        type=str,
+    ) = Field(
+        default=...,
+        description="The name of the exchange connector.",
+        client_data=ClientFieldData(
+            prompt=lambda mi: "Input your maker spot connector",
+            prompt_on_new=True,
+        ),
+    )
+    market: str = Field(
+        default=...,
+        description="The trading pair.",
+        client_data=ClientFieldData(
+            prompt=lambda mi: BaseTradingStrategyConfigMap.trading_pair_prompt(mi),
+            prompt_on_new=True,
+        ),
+    )
+
+    @classmethod
+    def trading_pair_prompt(cls, model_instance: 'BaseTradingStrategyConfigMap') -> str:
+        exchange = model_instance.exchange
+        example = AllConnectorSettings.get_example_pairs().get(exchange)
+        return (
+            f"Enter the token trading pair you would like to trade on"
+            f" {exchange}{f' (e.g. {example})' if example else ''}"
+        )
+
+    @validator("exchange", pre=True)
+    def validate_exchange(cls, v: str):
+        """Used for client-friendly error output."""
+        ret = validate_exchange(v)
+        if ret is not None:
+            raise ValueError(ret)
+        return v
+
+    @validator("market", pre=True)
+    def validate_exchange_trading_pair(cls, v: str, values: Dict):
+        exchange = values.get("exchange")
+        ret = validate_market_trading_pair(exchange, v)
+        if ret is not None:
+            raise ValueError(ret)
+        return v
+
+
+class BaseTradingStrategyMakerTakerConfigMap(BaseStrategyConfigMap):
+    maker_market: str = Field(
```

Comment:
This strategy base should also be adapted with the change

Reply:
I applied the change